### ISSUE TITLE
Feature request: rewording of allow buttons on block page

### ISSUE THREAD

**Issue by pgl** [uBlockOrigin/uBlock-issues#1681]
### Prerequisites

- [X] I verified that this is not a filter issue (MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues))
- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues) to avoid opening a duplicate issue
- [X] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### Description

I think it would be useful to consider changing the buttons that appear on the block page, to something more immediately obvious.

Currently the options to allow a block through reads like a sentence:

> Disable strict blocking for **example.com**

> Temporarily - Permanently

I can see the logic behind it and have explained it to people before. But I think it could be presented in a more user-friendly way, as simple buttons that permanently or temporarily allow, the rule. It would be nice to be able to say "just click permanently allow".

One other issue is that I've had to explain (and look up when I've forgotten) what "temporarily" means exactly - it's unclear whether this is for the single network request, or the browser session, or what. My understanding now is that it's for the length of the browser session.

So, my suggestion would be something like:

> For **example.com**:

> Allow for session - Permanently allow

Here's a quick mockup.

![image](https://user-images.githubusercontent.com/1181139/128334692-318e951d-b4e6-465e-9f18-d2293d13cf09.png)

I know there are some other considerations, but this I think this makes it more clear to the average end user what they're doing. Reading through the buttons and text as a sentence makes this clear, but unfortunately while scanning a block page it's not instantly clear.

### A specific URL where the issue occurs

https://doubleclick.net

### Steps to Reproduce

1. Go to a blocked URL

### Expected behavior

See very clear buttons with choices to allow blocked things for the session, or permanently.

### Actual behavior

See buttons which are a little bit unclear unless read carefully.

### uBlock Origin version

1.37.2

### Browser name and version

Chrome latest

### Operating System and version

macOS

**Comment by krystian3w** [comment]
> what "temporarily" means exactly - it's unclear whether this is for the single network request

for 120 seconds, so not session.

https://github.com/gorhill/uBlock/wiki/Advanced-settings#strictblockingbypassduration

**Comment by pgl** [comment]
Oh! I did not realise it was just 120 seconds. Well, I guess that shows that it's unclear even to someone who's relatively familiar with uBlock Origin. :) Perhaps we could reword it to say "Block for 120 seconds"? Or whatever - just something other than the rather vague "temporarily". (Absolutely not a criticism by the way, I just want to make it easier for people to understand what the buttons mean.)

**Comment by gorhill** [comment]
Something like "This one time", and strict-blocking would be disabled only for the current tab and until the site is navigated away to another origin?

**Comment by pgl** [comment]
@gorhill Would that require any functionality changes? It sounds ideal actually.

**Comment by gorhill** [comment]
That functionality is almost already there, the 120 seconds timer is actually a countdown: each time you navigate to a new page on the bypassed strict-blocked site, the countdown is reset to 120 seconds (configurable from advanced setting). I could increase the countdown to make it less likely that an idle page will cause strict-blocking to be unexpectedly re-enabled.

**Comment by pgl** [comment]
@gorhill Then I think that absolutely makes sense. Should we create another issue for that part? I think it would till be useful to change the wording and layout on the block page slightly.

Given the idea to change the "temporarily" button to "Allow this one time", perhaps we could change the other button to "Allow permanently"?

Updating the buttons to be actions, rather than part of a sentence, makes the intent more clear (IMO).

**Comment by felix-22** [comment]
Related https://github.com/uBlockOrigin/uBlock-issues/issues/77

**Comment by felix-22** [comment]
> gorhill : https://github.com/uBlockOrigin/uBlock-issues/issues/77#issuecomment-398776476 : A simpler solution is to disable strict-blocking until navigating away from the site, or maybe better, until the tab is closed.

> gorhill : https://github.com/uBlockOrigin/uBlock-issues/issues/1681#issuecomment-914291699 : Something like "This one time", and strict-blocking would be disabled only for the current tab and until the site is navigated away to another origin?

> gorhill : https://github.com/uBlockOrigin/uBlock-issues/issues/1746#issuecomment-932787794 : With #1681 I will probably remove the temporary bypass when the tab is closed.

Not a bad idea, more accurate/precise/flexible than a configurable timer, as an user never knows how much time he will spend on a temporary strict-unblocked site, it simply depends on the user, and the site.

<br>
<br>
<br>

> gorhill : until navigating away from the site, 

Some badware sites redirect an user away from a site, either automatically or after user's interaction, sometimes even multiple redirects

> gorhill : only for the current tab

Given we encounter a badware sites which does redirect, the sites do it in various ways: current tab, new tab, popup, popunder

> gorhill : or maybe better, until the tab is closed

Some badware websites prevents an user from closing a tab or from anvigating away

**Comment by gwarser** [comment]
![image](https://user-images.githubusercontent.com/886325/189502580-bd77804d-44c0-44eb-b2a0-3abfc3700d63.png)

**Comment by krystian3w** [comment]
Dark:

![obraz](https://user-images.githubusercontent.com/35370833/189502808-6f8612c3-95f4-4d6b-bf0f-316cd145bcf0.png)


Don't forget update wiki.

**Comment by krystian3w** [comment]
Maybe back works if tab have history to undo.

**Comment by gorhill** [comment]
This is Brave's version of the page:

![Screenshot from 2022-09-10 21-09-48](https://user-images.githubusercontent.com/585534/189507438-23c935ad-9ce6-4b3d-ab10-311323c1ad77.png)

**Comment by stephenhawk8054** [comment]
I'm a bit confused. If user wants to allow the site 1 time (or temporarily in the past), I guess it is clicking on `Proceed` button right? So, where should user clicks if they want to allow permanently?

**Comment by krystian3w** [comment]
may need now two clicks:

 check "don't warn" and now Proceed.

I see generated fine:

`no-strict-blocking: domain.tld true`

**Comment by stephenhawk8054** [comment]
I think that might cause some confusions for some. I have seen more than one time when some users asked how to hide the warning tab while still blocking that site (they think the tab is annoying, although we know it's not good to just "hide/close" the tab). They might understand clicking on the `Don't warn` tick will do exactly the above while not realizing they are allowing the site, not blocking it.

**Comment by gwarser** [comment]
> I'm a bit confused. If user wants to allow the site 1 time (or temporarily in the past), I guess it is clicking on `Proceed` button right? So, where should user clicks if they want to allow permanently?

> may need now two clicks...
> 
> check "don't warn" and now Proceed.

People may expect this will also work for "Go back" :confused: 

***

Compare to Brave, main header seems to be missing :)

**Comment by ghajini** [comment]
l little bit offcentred compared to old one
a) old one

![image](https://user-images.githubusercontent.com/20338483/189522127-1f5bf3c4-88fb-46ff-943c-8738803358bc.png)

b)new one
![firefox_6BuFJKT3yc](https://user-images.githubusercontent.com/20338483/189522250-c7429f06-3caf-4ce7-b8c7-37fee1f56128.png)

**Comment by pgl** [comment]
Can I just say that, design aside, the rewording here is amazing!

**Comment by gorhill** [comment]
> Compare to Brave, main header seems to be missing :)

Yes, but I can't use that sentence in uBO, the document might be blocked for other reasons than just tracking.

> That orange color in the old one grabbed more attention than blue.

Yes, I think this was wrong in retrospect, too many people throught this was a virus-level warning. Just a plain warning is fine.

> Can I just say that, design aside, the rewording here is amazing!

Well, it's essentially Brave's design.

**Comment by gorhill** [comment]
> People may expect this will also work for "Go back" confused

Maybe I could disable Go back/Close window buttons when the box is checked?
